address/book.go

```diff
@@ -193,15 +194,70 @@ func NewBook(cfg BookConfig) *Book {
 	}
 }
 
+// queryAssetInfo attempts to locate asset genesis information by querying
+// geneses already known to this node. If asset issuance was not previously
+// verified, we then query universes in our federation for issuance proofs.
+func (b *Book) queryAssetInfo(ctx context.Context,
+	id asset.ID) (*asset.AssetGroup, error) {
+
+	// Check if we know of this asset ID already.
+	assetGroup, err := b.cfg.Store.QueryAssetGroup(ctx, id)
+	switch {
+	case assetGroup != nil:
+		return assetGroup, nil
+
+	// Asset lookup failed gracefully; continue to asset lookup using the
+	// AssetSyncer if enabled.
+	case errors.Is(err, ErrAssetGroupUnknown):
+		if !b.cfg.SyncerActive {
+			return nil, ErrAssetGroupUnknown
+		}
+
+	case err != nil:
+		return nil, err
+	}
+
+	log.Debugf("asset %v is unknown, attempting to bootstrap", id.String())
+
+	// Use the AssetSyncer to query our universe federation for the asset.
+	err = b.cfg.Syncer.FetchAssetInfo(ctx, &id)
+	if err != nil {
+		return nil, err
+	}
+
+	// The asset genesis info may have been synced from a universe
+	// server; query for the asset ID again.
+	assetGroup, err = b.cfg.Store.QueryAssetGroup(ctx, id)
```

Comment:
Could we have `Syncer.FetchAssetInfo` also return some results above? Could save the extra call & store roundtrip

Reply:
Yes, but contingent on doing the group lookup on the multiverse side / when looking up the root node itself.